[#20] 게시물, 댓글, 좋아요, 좋아요 클릭 유저 Entity 생성

choitaehoon opened the pull request:
- BaseTimeEntity class 생성하여 생성 날짜, 수정 날짜 자동으로 변경

choitaehoon committed:
[#11] 게시물 작성

- 게시물 작성 구현
- AccountController, LoginController naming 변경

choitaehoon committed:
[#14] AWS S3 환경설정

- AWS의 key값 분리 후 gitignore 적용
- AmazonS3 빈으로 등록

choitaehoon committed:
[#14] AWS S3 환경설정

- javax.annotation-api 종속성 추가

choitaehoon committed:
[#14] AWS S3 환경설정

- PropertySource 제거 후 EnvironmentPostProcessor 생성
- aws 환경설정 예외 처리
- aws 환경 설정, 메시지에 관련된 상수 생성

choitaehoon committed:
[#14] AWS S3 환경설정

- gitignore application-aws.yml로 변경

choitaehoon committed:
[#14] AWS S3 환경설정

- 주석제거

choitaehoon committed:
[#14] AWS S3 환경설정

- EnvironmentPostProcessor 제거
- application.yml 파일안에 aws include

choitaehoon committed:
[#16] Aws S3 이미지 업로드

- AWS S3 이미지 업로드 구현
- 파일 예외 처리

choitaehoon committed:
[#16] Aws S3 이미지 업로드

- FileController API naming 변경
- upload 상수로 반환
- 에러 로그 추가
- AWS S3 인터페이스로 확장

choitaehoon committed:
[#20] 게시물에 대한 전반적인 도메인 생성

- 게시물, 댓글, 좋아요, 좋아요 클릭 유저 Entity 생성
- BaseTimeEntity class 생성하여 생성 날짜, 수정 날짜 자동으로 변경

f-lab-dev commented on src/main/java/com/social/instagram/domain/Reply.java:
```diff
@@ -0,0 +1,30 @@
+
+@Entity
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
```
엔티티마다 보니까 전체 생성자가 없는데 그러면 새로 객체를 생성할땐 어떻게 생성할까요?

choitaehoon replied:
일단 엔티티 설계 작업이므로 칼럼 변경이 잦아 생성자를 생성하지 않았고 이 엔티티 설계가 문제가 없다면 생성자를 만들겠습니다.

f-lab-dev replied:
그러면 이것은 작업중인것인가요? 아니면 완료되어 리뷰를 필요로 하는 작업인가요?

choitaehoon replied:
리뷰를 통해 설계 과정에 대한 검토를 받으려했습니다. 일단 제가 생각하기에 설계에 대한 문제가 없다고 판단하여 구현하고 올리겠습니다

f-lab-dev commented on src/main/java/com/social/instagram/domain/time/BaseTimeEntity.java:
```diff
@@ -0,0 +1,30 @@
+@Getter
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener.class)
+public class BaseTimeEntity {
```
👍

choitaehoon committed:
[#16] AWS S3 이미지 업로드

- AwsS3Config 안에 bucket 빈으로 등록
- 예외 에러 로그 추가
- AwsS3Connector에서 AwsS3LocalService로 upload 옮기기

choitaehoon committed:
[#20] 게시물에 대한 전반적인 도메인 생성

- Post 클래스 fileName 제거

choitaehoon committed:
[#16] AWS S3 이미지 업로드

- ExceptionHandler를 이용하여 예외 객체 로그 적용
- upload 로그 debug로 낮추기

choitaehoon committed:
[#20] 게시물에 대한 전반적인 도메인 생성

- 생성자 생성

choitaehoon committed:
[#16] AWS S3 이미지 업로드

- Post Entity에 filePath 추가
- S3 이미지 업로드 후 Post Entity에 filePath, userId 저장

choitaehoon committed:
[#16] AWS S3 이미지 업로드

- PostEntity로 변경하는 메소드 naming 변경

f-lab-dev commented:
이거는 계속 바뀌고있는 것 같은데 최종본이 나오면 리뷰요청 한번 해주세요~

choitaehoon committed:
[#16] AWS S3 이미지 업로드

- 디버깅에 용이하게 만들고자 ExceptionHandler안에 stacktrace 출력 구현

f-lab-dev commented:
그리고 `게시물에 대한 전반적인 도메인 생성`라는 말은 작업내용이 명확하지 않아보입니다. 정확히 어떤 작업을 했는지 명확하게 제목에 표현할 수 있도록 고민해주세요~

choitaehoon committed:
[#16] AWS S3 이미지 업로드

- Exception Handler error 로그 수정
- AwsS3LocalService @Value 생성자 주입
- AwsS3Config bucket 빈 삭제

choitaehoon committed:
[#16] AWS S3 이미지 업로드

- ExceptionUtils 의존성 삭제

choitaehoon committed:
Merge branch 'develop' into feature/16-aws-upload

choitaehoon committed:
Merge branch 'feature/16-aws-upload' into feature/20-post-and-nice-domain
